cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerLdapService.java

```diff
@@ -34,10 +35,22 @@
     @Inject
     private ClouderaManagerClientFactory clouderaManagerClientFactory;
 
+    @Inject
+    private GrpcUmsClient umsClient;
+
     public void setupLdap(Stack stack, Cluster cluster, HttpClientConfig clientConfig) throws ApiException {
         LdapConfig ldapConfig = cluster.getLdapConfig();
         ClouderaManagerResourceApi clouderaManagerResourceApi = clouderaManagerClientFactory.getClouderaManagerResourceApi(stack, cluster, clientConfig);
-        clouderaManagerResourceApi.beginTrial();
+
+        // Begin the Cloudera Manager trial only if UMS is not enabled. Otherwise, we'll be using a
+        // license from UMS.
+        if (!umsClient.isUmsUsable(stack.getCreator().getUserCrn())) {
```

Comment:
What's the usecase for trial? I can't see the relation with the isUmsUsable condition.

Reply:
Trial is useful when you don't have a license but you want to try out enterprise features. When using UMS, CM is provisioned with a full license, so there's no need to enable the trial.

Reply:
On QA env the QAAS team can use the trial method as we don't have a proper UMS to provide a licence.

Reply:
Would it be better to make enabling trial a feature flag? Is it useful outside of the testing context?

Reply:
I don't think we need to take care of that at the moment since we're not going to ship CB with CM to customers outside of Altus. What would be better to extend CB API to take the licence, but it's not important at the moment.